### PR TITLE
Local node setup logic

### PR DESCRIPTION
**Pull request by yhaspel**

### CODE REVIEW

File: app/infra/diskStorageService/diskStorageService.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// @flow
+import { ipcRenderer } from 'electron';
+import { ipcConsts } from '/vars';
+
+const getBytesfromGb = (Gb: number) => Gb * 1073741824;
+const LEAVE_AVAILABLE_SPACE = 50; // GB
```
Comment (IlyaVi) [inline app/infra/diskStorageService/diskStorageService.js]: give a better name to this var

File: app/screens/localNode/LocalNodeLoadingPage.js [hunk 2]
```diff
@@ -59,6 +61,10 @@ class LocalNodeLoadingPage extends Component<Props, State> {
     }, 10000);
   }
 
+  componentWillUnmount() {
+    this.timer && clearTimeout(this.timer);
```
Comment (IlyaVi) [inline app/screens/localNode/LocalNodeLoadingPage.js]: where do you set this.timer? In this version it is always undefined

File: app/screens/localNode/LocalNodeSetupPage.js [hunk 3]
```diff
@@ -2,74 +2,47 @@
 import React, { Component } from 'react';
 import styled from 'styled-components';
 import { connect } from 'react-redux';
-import { setLocalNodeStorage } from '/redux/localNode/actions';
+import { setLocalNodeStorage, getDrivesList, getAvailableSpace } from '/redux/localNode/actions';
 import { BaseText, BoldText, LeftPaneInner, GrayText, LocalNodeBase, RightPaneSetup } from '/components/localNode';
 import { SmButton, SmDropdown } from '/basicComponents';
 import type { DropdownEntry } from '/basicComponents';
 import type { Action } from '/types';
+// import diskStorageService from '/infra/diskStorageService';
```
Comment (IlyaVi) [inline app/screens/localNode/LocalNodeSetupPage.js]: remove comment

File: app/screens/localNode/LocalNodeSetupPage.js [hunk 4]
```diff
@@ -2,74 +2,47 @@
...
+
+type SetupPageProps = {
+  history: any,
+  // eslint-disable-next-line react/no-unused-prop-types
```
Comment (IlyaVi) [inline app/screens/localNode/LocalNodeSetupPage.js]: why disabling eslint? This prop is used later

File: desktop/main.dev.js [hunk 5]
```diff
@@ -14,6 +14,7 @@ import log from 'electron-log';
 import { ipcConsts } from '../app/vars';
 import MenuBuilder from './menu';
 import FileManager from './fileManager';
+import DiskStorageManager from './DiskstorageManager';
```
Comment (IlyaVi) [inline desktop/main.dev.js]: file name is diskStorageManager so import will fail, did you tested it?

File: desktop/diskStorageManager.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+type Drive = {
+  id: string,
+  mountPoint: string,
+  label: string
+};
+
+class DiskStorageManager {
+  static getDriveList = async ({ event }: { event: any }) => {
+    drivelist.list((error, drives) => {
```
Comment (IlyaVi) [inline desktop/diskStorageManager.js]: you're stating that the function is async - then await usage is expected to async call synchronous but the code is written using callback and no sign of Promise whatsoever...

File: app/screens/localNode/LocalNodeLoadingPage.js (hunk 2 above)
Comment (yhaspel) [reply app/screens/localNode/LocalNodeLoadingPage.js]: error in merge. fixed.

File: desktop/main.dev.js (hunk 5 above)
Comment (yhaspel) [reply desktop/main.dev.js]: mistake. fixed.

File: desktop/diskStorageManager.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+type Drive = {
+  id: string,
+  mountPoint: string,
+  label: string
+};
+
+class DiskStorageManager {
+  static getDriveList = ({ event }: { event: any }) => {
+    drivelist.list((error, drives) => {
```
Comment (IlyaVi) [inline desktop/diskStorageManager.js]: the docs of the library states that main usage is through async function

```
const drivelist = require('drivelist');

const drives = await drivelist.list();
console.log(drives);

```

File: app/screens/localNode/LocalNodeSetupPage.js [hunk 8]
```diff
@@ -172,23 +149,30 @@ class LocalNodeSetupPage extends Component<Props, State> {
...
-  handleSelectDrive = ({ index }: { index: number }) => this.setState({ selectedDriveIndex: index });
+  handleSelectDrive = ({ index }: { index: number }) => {
+    const { getAvailableSpace, drives } = this.props;
+    const drive: any = drives && drives[index];
+    drive && getAvailableSpace(drive.mountPoint);
+    this.setState({ selectedDriveIndex: index });
+  };
 
-  handleSelectCapacity = ({ index }: { index: number }) => this.setState({ selectedCapacityIndex: index });
+  handleSelectCapacity = ({ index }: { index: number }) => {
```
Comment (IlyaVi) [inline app/screens/localNode/LocalNodeSetupPage.js]: revert

File: app/redux/localNode/reducer.js [hunk 9]
```diff
@@ -1,18 +1,32 @@
...
+      const {
+        payload: { drives }
+      } = action;
+      return { ...state, drives };
+    }
+    case GET_AVAILABLE_DISK_SPACE: {
+      const {
+        payload: { availableDiskSpace, capacities }
+      } = action;
+      return { ...state, availableDiskSpace, capacities };
```
Comment (IlyaVi) [inline app/redux/localNode/reducer.js]: rename "capacities" to a more explaining name to differ from "capacity" and add it to initial state

File: app/screens/localNode/LocalNodeSetupPage.js [hunk 10]
```diff
@@ -97,26 +67,27 @@ const GrayTextWrapper = styled.div`
...
   };
 
   render() {
     return <LocalNodeBase header="Local Node Setup" leftPane={this.renderLeftPane()} rightPane={this.renderRightPane()} />;
   }
 
+  componentDidMount() {
+    const { getDrivesList, drive, capacity, drives, capacities } = this.props;
+    getDrivesList();
+    const timer = setTimeout(() => {
```
Comment (IlyaVi) [inline app/screens/localNode/LocalNodeSetupPage.js]: WTF???

File: app/screens/localNode/LocalNodeSetupPage.js [hunk 11]
```diff
@@ -172,23 +149,30 @@ class LocalNodeSetupPage extends Component<Props, State> {
...
+    const { setLocalNodeStorage, history, drives, capacities } = this.props;
+    const { selectedCapacityIndex, selectedDriveIndex } = this.state;
+    setLocalNodeStorage({ capacity: capacities[selectedCapacityIndex], drive: drives[selectedDriveIndex] });
     history.push('/main/local-node/local-node-loading');
   };
 
-  handleSelectDrive = ({ index }: { index: number }) => this.setState({ selectedDriveIndex: index });
+  handleSelectDrive = ({ index }: { index: number }) => {
+    const { getAvailableSpace, drives } = this.props;
+    const drive: any = drives && drives[index];
```
Comment (IlyaVi) [inline app/screens/localNode/LocalNodeSetupPage.js]: useless check - no data - no selections

File: desktop/diskStorageManager.js (hunk 7 above)
Comment (yhaspel) [reply desktop/diskStorageManager.js]: There's an issue with version 8+ of this package. Version 6 is installed works as coded.

Comment (yhaspel) [reply desktop/diskStorageManager.js]: The issue is with electron and version 8 of this package.

File: app/screens/localNode/LocalNodeSetupPage.js (hunk 11 above)
Comment (yhaspel) [reply app/screens/localNode/LocalNodeSetupPage.js]: ok

File: app/screens/localNode/LocalNodeSetupPage.js (hunk 10 above)
Comment (yhaspel) [reply app/screens/localNode/LocalNodeSetupPage.js]: No harm in leaving a proper comment.

File: app/redux/localNode/reducer.js (hunk 9 above)
Comment (yhaspel) [reply app/redux/localNode/reducer.js]: ok

**Pull request merged**